### PR TITLE
making a few tweaks so matrices > 30 do not break the app

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
Making a few small tweaks such that larger matrices (> 30) do not appear like they are breaking the application.
This PR has two things
- If matrix is too large then don't display the tooltip preview
- Set minimum dimension in model-diagram so it doesn't look like the matrix/model is broken

image

image

### Testing
Can try with this AMR model 
[75.json](https://github.com/user-attachments/files/17496367/75.json)

### CODE REVIEW

File: packages/client/hmi-client/src/components/model/petrinet/model-configurations/tera-stratified-matrix-preview.vue [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <template>
 	<section>
-		<section class="matrix">
+		<div v-if="matrix.length > 20 || matrix[0].length > 20">Matrix is {{ matrix.length }}x{{ matrix[0].length }}</div>
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/model/petrinet/model-configurations/tera-stratified-matrix-preview.vue:3]: The 20 is kind of arbitrary, we probably want to adjust it when we have better guidance on how people would interact with this matrices in terarium.

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/model/petrinet/model-configurations/tera-stratified-matrix-preview.vue:3]: nitpick - can we have this 20 stored in a `const maxDisplayDimension` (whatever var name you want Im not picky)

**Review by Tom-Szendrey — APPROVED** [review]

File: packages/client/hmi-client/src/components/model/petrinet/model-configurations/tera-stratified-matrix-preview.vue (hunk 1 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/components/model/petrinet/model-configurations/tera-stratified-matrix-preview.vue]: done

File: packages/client/hmi-client/src/model-representation/petrinet/nested-petrinet-renderer.ts [hunk 2]
```diff
@@ -229,17 +229,18 @@ export class NestedPetrinetRenderer extends PetrinetRenderer {
...
+						const w = Math.max(d.width / matrixColLen);
+						const h = Math.max(d.height / matrixRowLen);
+
 						transitionNode
 							.append('rect')
-							.attr('width', d.width / matrixColLen)
-							.attr('height', d.height / matrixRowLen)
+							.attr('width', w)
+							.attr('height', h)
 							.attr('x', -d.width * 0.5 + (d.width / matrixColLen) * colIdx)
```
Comment (dgauldie) [inline packages/client/hmi-client/src/model-representation/petrinet/nested-petrinet-renderer.ts:239]: can this be w and h as well then?

File: packages/client/hmi-client/src/components/model/petrinet/model-configurations/tera-stratified-matrix-preview.vue [hunk 3]
```diff
@@ -46,6 +49,7 @@ const subject = ref('');
 const outcome = ref('');
 
 const matrixColLen = computed(() => matrix.value[0]?.length ?? 0);
+const matrixThreshold = 20;
```
Comment (dgauldie) [inline packages/client/hmi-client/src/components/model/petrinet/model-configurations/tera-stratified-matrix-preview.vue:52]: Do we not usually have a convention of all upper case letters for const primitives? It's ok if not. Just asking

File: packages/client/hmi-client/src/model-representation/petrinet/nested-petrinet-renderer.ts [hunk 4]
```diff
@@ -229,17 +229,18 @@ export class NestedPetrinetRenderer extends PetrinetRenderer {
 				row.forEach((col, cidx) => {
 					const colIdx = cidx;
 					if (col.content && col.content.value) {
+						const w = Math.max(d.width / matrixColLen);
```
[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/client/hmi-client/src/model-representation/petrinet/nested-petrinet-renderer.ts:232]: Interesting ... I am pretty sure I wrote it as

```
Math.max(expression, 1) 
```

to ensure at least 1-pixel width

[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/client/hmi-client/src/model-representation/petrinet/nested-petrinet-renderer.ts:232]: Actually there is a better fix here ...

File: packages/client/hmi-client/src/components/model/petrinet/model-configurations/tera-stratified-matrix-preview.vue (hunk 3 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/components/model/petrinet/model-configurations/tera-stratified-matrix-preview.vue]: fixed

**Review by shawnyama — APPROVED** [review]

**Pull request merged**